src/component/panels/IntegralsPanel/IntegralsPreferences.tsx

```diff
@@ -84,63 +79,51 @@ function IntegralsPreferences(props, ref) {
     },
     [preferences],
   );
+  const { handleSubmit, control } = useForm<any>({
+    defaultValues: preferencesByNuclei,
+  });
 
-  useImperativeHandle(ref, () => ({
-    saveSetting: () => {
-      formRef.current.submitForm();
-    },
-  }));
+  useSettingImperativeHandle(ref, handleSubmit, saveHandler);
```

Comment:
You probably don't need to do this. If it's because the submit button is not inside the form, there is a way to link it using an id: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#form

Reply:
The button being outside the form is one reason, but I also need to check the validation of the form before closing the preferences panel